### ISSUE TITLE
[FTBFS] julia - need to specify CPU for openblas?

### ISSUE THREAD

**Issue by bobertlo** [void-linux/void-packages#34387]
I am trying to do a revbump on libgit2 and julia is the only build not working. I checked out master and it is not able to rebuild from source.

Error:
```
getarch_2nd.c:14:35: error: 'SGEMM_DEFAULT_UNROLL_M' undeclared (first use in this function); did you mean 'SHGEMM_DEFAULT_UNROLL_M'?
   14 |     printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   SHGEMM_DEFAULT_UNROLL_M
```

It looks like this error: https://github.com/xianyi/OpenBLAS/issues/2769

I will try to look into this further when I have time. I am also wondering if this should be specified anyway are we generating code specific to the host/builder CPU microarchitecture?

**Comment by ericonr** [comment]
https://github.com/void-linux/void-packages/pull/33535 isn't failing with that, I think.

What machine are you developing on? If that's to blame, we really do need to fix the package so it doesn't leak builder details.

**Comment by bobertlo** [comment]
It’s failing in both glibc and musl vms on a i7-10750H. It appears to be cpu detection in the auto config and threads with this error mentioned newer cpus not being detected properly yet.

iirc openblas has been updated since that pr or near then but I’m guessing it is a host info leak. I’ll try it natively when I am near that system.

**Comment by bobertlo** [comment]
I was able to build on a 2nd get i7 in native glibc.

I did a little more research:

https://linux.debian.bugs.dist.narkive.com/ojoRHuKs/bug-923607-openblas-ftbfs-if-cpu-is-not-detected

According to this there are many microarchitecture kernels built and selected at runtime. I am unsure why it is detecting the host cpu information then, but it does not look like the host information is leaking into the build as much as breaking the build process.
